### PR TITLE
[Frontend] Adding line offset for code problems

### PR DESCRIPTION
**Pull request by AlexandreDoneux**
This PR adds the possibility through the task edit form to have a line offset for the code blocks in code problems. Adding a field to the form, checking it's value and applying the changes to the particular problem. An error will be displayed if the user inputs something other than a number.

### CODE REVIEW

File: inginious/frontend/static/js/common.js
Comment (anthonygego) [inline inginious/frontend/static/js/common.js:13]: Maybe you can try this instead of the `parseInt` + `isNaN` check below.

```suggestion
        registerCodeEditor(elem, $(elem).attr('data-x-language'), $(elem).attr('data-x-lines'), $(elem).attr('data-x-first-line') || 1);
```

**Review by anthonygego — CHANGES_REQUESTED** [review]

File: inginious/common/tasks_problems.py
Comment (anthonygego) [inline inginious/common/tasks_problems.py:192]: Beware here, because you return the result of the parent class method only, so if you want to store `offset` as a integer in the task descriptor, this won't do the job.

File: inginious/frontend/pages/course_admin/task_edit.py
Comment (anthonygego) [inline inginious/frontend/pages/course_admin/task_edit.py:145]: This is a bit pedantic but to keep PRs the cleanest and easiest to review, please avoid to commit files with no useful changes.

I strongly recommend you to give a first review of your PR by yourself before submitting it for review. This is also useful to when debugging lines remain: there is no chance the PR will be merged in those cases so that automatically requires a second review.

File: inginious/common/tasks_problems.py
Comment (anthonygego) [inline inginious/common/tasks_problems.py:185]: Here you will skip the check if `problem_content["offset"]` is an empty string, that is, if the field is left empty or as is. Empty string will override the default `1` value when loaded  in the `CodeProblem` constructor.

This will lead to an empty string being passed through CodeMirror and a first line beginning at `0`... you should either save the default value, or remove the empty string from the settings dict in this function.

**Comment by anthonygego** [comment]
Can you rebase this PR so that only your commits appear here ? Otherwise it's pretty hard to review.

**Comment by AlexandreDoneux** [comment]
Rebase on `2e692aa7`.

File: inginious/common/tasks_problems.py
Comment (anthonygego) [inline inginious/common/tasks_problems.py:186]:
```suggestion
            del problem_content["offset"]
```

This ease YAML file reading and saves some space.

**Review by anthonygego — APPROVED** [review]

**Pull request merged**
